app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.java

```diff
@@ -0,0 +1,262 @@
+import de.westnordost.streetcomplete.quests.AbstractQuestFormAnswerFragment;
+import de.westnordost.streetcomplete.view.dialogs.AlertDialogBuilder;
+
+public class AddMaxHeightForm extends AbstractQuestFormAnswerFragment
+{
+	public static final String
+		MAX_HEIGHT = "max_height",
+		NO_SIGN = "no_sign";
+
+	public static final int
+		IS_BELOW_DEFAULT = 1,
+		IS_DEFAULT = 2,
+		IS_NOT_INDICATED = 3;
+
+	private EditText heightInput, feetInput, inchInput;
+	private Spinner heightUnitSelect;
+
+	View meterInputSign, feetInputSign;
+
+	@Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
+									   Bundle savedInstanceState)
+	{
+		View view = super.onCreateView(inflater, container, savedInstanceState);
+
+		String unit = getCountryInfo().getMeasurementSystem().get(0).equals("metric") ? Height.METRIC : Height.IMPERIAL;
+		setMaxHeightSignLayout(R.layout.quest_maxheight, unit);
+		addOtherAnswers();
+
+		return view;
+	}
+
+	private void setMaxHeightSignLayout(int resourceId, String unit)
+	{
+		View contentView = setContentView(getCurrentCountryResources().getLayout(resourceId));
+
+		heightInput = contentView.findViewById(R.id.meterInput);
+		feetInput = contentView.findViewById(R.id.feetInput);
+		inchInput = contentView.findViewById(R.id.inchInput);
+
+		meterInputSign = contentView.findViewById(R.id.meterInputSign);
+		feetInputSign = contentView.findViewById(R.id.feetInputSign);
+
+		heightUnitSelect = contentView.findViewById(R.id.heightUnitSelect);
+		initHeightUnitSelect();
+
+		switchLayout(unit);
+	}
+
+	private void switchLayout(String unit)
+	{
+		if (unit.equals(Height.METRIC))
+		{
+			meterInputSign.setVisibility(View.VISIBLE);
+			if (feetInputSign != null)
+			{
+				feetInputSign.setVisibility(View.GONE);
+			}
+		} else {
+			feetInputSign.setVisibility(View.VISIBLE);
+			if (meterInputSign != null)
+			{
+				meterInputSign.setVisibility(View.GONE);
+			}
+		}
+	}
+
+	private void initHeightUnitSelect()
+	{
+		List<String> measurementUnits = getCountryInfo().getMeasurementSystem();
+		heightUnitSelect.setVisibility(measurementUnits.size() == 1 ? View.GONE : View.VISIBLE);
+		heightUnitSelect.setAdapter(new ArrayAdapter<>(getContext(), R.layout.spinner_item_centered, getSpinnerItems(measurementUnits)));
+		heightUnitSelect.setSelection(0);
+
+		heightUnitSelect.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+			@Override
+			public void onItemSelected(AdapterView<?> parentView, View selectedItemView, int position, long id)
+			{
+				String heightUnit = heightUnitSelect.getSelectedItem().equals("m") ? Height.METRIC : Height.IMPERIAL;
+				switchLayout(heightUnit);
+			}
+
+			@Override
+			public void onNothingSelected(AdapterView<?> parentView) {}
+		});
+	}
+
+	private List<String> getSpinnerItems(List<String> units)
+	{
+		List<String> items = new ArrayList<>();
+
+		for (int i = 0; i < units.size(); i++) {
+			String unit = units.get(i);
+			if (unit.equals("metric"))
```

Comment:
Shouldn't you use the constants here?

Reply:
Well, but they don't exist anymore if I'm using a enum there 😄